README.md

```diff
@@ -17,57 +15,147 @@ Widgets are reusable building blocks used to create complex and configurable use
+
+```php
+/**
+ * @var \Psr\Container\ContainerInterface $container
+ */
+
+\Yiisoft\Widget\WidgetFactory::initialize($container);
+```
+
+The factory initialization must be performed only once. It is a good idea to do that for the whole application.
+See Yii example in the configuration file of this package
+[`config/providers.php`](https://github.com/yiisoft/widget/blob/master/config/providers.php).
+
+### Configuring the widget
+
+You can configure the widget when creating its instance, for example,
+the widget class must accept some ID when initializing the object.
+
+```php
+final class MyWidget extends \Yiisoft\Widget\Widget
+{
+    private string $id;
+
+    public function __construct(string $id)
+    {
+        $this->id = $id;
+    }
+
+    protected function run(): string
+    {
+        return $this->id;
+    }
+}
+```
+
+To set a value for the ID, you can pass it in the configuration array to the `widget()` method:
 
 ```php
-<?= MyWidget::widget()->options(['class' => 'testMe']) ?>
+<?= MyWidget::widget([
+    '__construct()' => [
+        'id' => 'value',
+    ],
+]) ?>
 ```
 
-`options()` is an example method. There could be many methods that configure a widget.
-Note that these configuration methods are returning a clone of the widget instance. 
+For a description of the configuration syntax, see the [yiisoft/factory](https://github.com/yiisoft/factory)
+package documentation.
+
+### Widget for capturing content
 
-A special type of widgets, wrapping widgets, mimic HTML opening and closing tags wrapping some content.
+Some widgets can take a block of content which should be enclosed between the invocation of `begin()` and `end()`
+methods. These are wrapping widgets that mimic opening and closing HTML tags that wrap some content.
 They are used a bit differently:
 
 ```php
-<?= MyWidget::widget()->options(['class' => 'testMe'])->begin() ?>
+<?= MyWidget::widget()->begin() ?>
     Content
 <?= MyWidget::end() ?>
 ```
 
-The package ensures that all these are properly opened, closed and nested.
+For your widget to do this, you need override the parent `begin()` method and don't forget to call `parent::begin()`.
+For example like this:
 
-## Implementing your own widgets
+```php
+final class MyWidget extends \Yiisoft\Widget\Widget
+{
+    public function begin(): ?string
+    {
+        parent::begin();
+        ob_start();
+        ob_implicit_flush(false);
+        return null;
+    }
+
+    protected function run(): string
+    {
+        return (string) ob_get_clean();
+    }
+}
+```
 
-In order to implement your own widget, extend from `\Yiisoft\Widget\Widget`. If it is a regular widget, it is enough
-to implement `run()` method.
+The package ensures that all widgets are properly opened, closed and nested. For a description of all the methods that
+can be overridden, see the [`Yiisoft\Widget\Widget`](https://github.com/yiisoft/widget/blob/master/src/Widget.php) class.
```

Comment:
We've already described `begin()` and `run()`. There are only `beforeRun()` and `afterRun()` left. Worth describing these in the readme as well rather than sending reader to the code.